File: src/llm.js
/**
 * LLM interaction logic using configured provider.
 * @module llm
 */

import { config } from "./config.js";
import { callOpenRouter } from "./providers/openrouter.js";
import { callGemini } from "./providers/gemini.js";

/**
 * Calls the LLM with the given messages and tools.
 * @param {Array} messages - The conversation history.
 * @param {Array} tools - The available tools.
 * @param {string} [model] - Optional model override.
 * @returns {Promise<Object>} The response message.
 */
export async function callLLM(messages, tools, model = null) {
  if (config.provider === "gemini") {
    return await callGemini(messages, tools, model);
  } else {
    return await callOpenRouter(messages, tools, model);
  }
}

File: src/llm.py
"""LLM provider router."""

from .config import config
from .providers.gemini import call_gemini
from .providers.openrouter import call_openrouter


async def call_llm(messages: list, tools: list, model: str = None) -> dict:
    """Route LLM calls to the configured provider."""
    if config.provider == "gemini":
        return await call_gemini(messages, tools, model)
    return await call_openrouter(messages, tools, model)
